enginetest/queries/query_plans.go

```diff
@@ -13379,7 +13379,7 @@ WHERE
 			" │                       │   ├─ name: E2I7U\n" +
 			" │                       │   └─ columns: [id dkcaj kng7t tw55n qrqxw ecxaj fgg57 zh72s fsk67 xqdyt tce7a iwv2h hpcms n5cc2 fhcyt etaq7 a75x7]\n" +
 			" │                       └─ HashLookup\n" +
-			" │                           ├─ left-key: TUPLE(e2i7u.id:0!null)\n" +
+			" │                           ├─ left-key: TUPLE(e2i7u.id:18!null)\n" +
```

Comment:
i think it's definitely correct to pick the column from inside the subquery. i added another test. If you think about it, the only way this can happen is if you have a table+column select inside a subquery that is the same table as the outside and does not decorrelate. It's kind of confusing to think about, the nested lookup doesn't really add anything logically. I'm having a hard time coming up with an example where the indexing choice creates a differing result.